src/main/scala/com/microsoft/hyperspace/index/sources/interfaces.scala

```diff
@@ -222,13 +201,54 @@ trait FileBasedSourceProvider extends SourceProvider {
   def getRelation(plan: LogicalPlan): Option[FileBasedRelation]
 
   /**
-   * Returns enriched index properties.
+   * Returns true if the given relation metadata is supported relation metadata.
    *
-   * @param relation Logical relation to retrieve necessary information.
-   * @param properties Index properties to enrich.
-   * @return Updated index properties for index creation or refresh.
+   * @param metadata Relation metadata to check if it's supported.
+   * @return Some(true) if the given relation metadata is supported relation metadata, otherwise
+   *         None.
+   */
+  def isSupportedRelationMetadata(metadata: Relation): Option[Boolean]
+
+  /**
+   * Returns the [[FileBasedRelationMetadata]] that wraps the given relation metadata
+   * if the given relation metadata is supported relation metadata
+   *
+   * @param metadata Relation metadata to wrap to [[FileBasedRelationMetadata]]
+   * @return [[FileBasedRelationMetadata]] that wraps the given relation metadata
+   */
+  def getRelationMetadata(metadata: Relation): Option[FileBasedRelationMetadata]
+}
+
+/**
+ * ::Experimental::
+ * A trait that represents relation metadata for a source provider.
+ *
+ * @since 0.5.0
+ */
+trait SourceRelationMetadata
+
+/**
+ * ::Experimental::
+ * A trait that a source provider should implement to represent the source relation metadata
+ * that is based on files.
+ *
+ * @since 0.5.0
+ */
+trait FileBasedRelationMetadata extends SourceRelationMetadata {
+
+  /**
+   * @return new [[Relation]] metadata that will have the latest source
+   */
+  def refresh(): Relation
+
+  /**
+   * @return file format name to read internal data
+   */
+  def internalFileFormatName(): String
+
+  /**
+   * @param properties Index properties to enrich
+   * @return updated index properties for index creation or refresh
```

Comment:
nit: same for other comments
```suggestion
   * @return Updated index properties for index creation or refresh.
```

Reply:
Actually [the official Javadoc guideline](https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html) (by which [the Scaladoc guide](https://docs.scala-lang.org/style/scaladoc.html) follows) states that after `@return` it should start with lowercase. Although it is not consistent, but many comments in Hyperspace follow this rule.

The guide also says that period should be used if the phrase is followed by another phrase or sentence.

https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html#@param:

> Parameter names are lowercase by convention. The data type starts with a lowercase letter to indicate an object rather than a class. The description begins with a lowercase letter if it is a phrase (contains no verb), or an uppercase letter if it is a sentence. End the phrase with a period only if another phrase or sentence follows it.

I think it makes sense because you don't end bullet list items with periods unless each item is a sentence.

Reply:
That's just a project convention & for consistency. I guess we follow several convention rules from apache/spark repository.

Reply:
Okay. Quick grepping (only for `@param` and `@return`) in Hyperspace shows:
- uppercase : lowercase = 7 : 3
- period : no period = 8 : 2

So although it's not super consistent, it seems the tendency is clear here.

I respect the existing rule and like consistency, so I'll follow it here.

However, in Spark, the tendency is different:
- uppercase : lowercase = 4 : 6
- period : no period = 4 : 6

Personally I have no problem with uppercase but the period thing bugs me a little, because a period isn't really for a phrase; it's for a proper sentence.

Reply:
yea I think the period thing is reasonable. @imback82 WDYT?

Reply:
I am fine with either one as long as we are consistent about it in this repo. If we are changing the prominent usage (e.g., going from period to no-period), let's do it as a separate PR to fix it all.